src/edu/washington/escience/myria/expression/evaluate/PythonUDFEvaluator.java

```diff
@@ -97,22 +100,17 @@ public PythonUDFEvaluator(
    */
   private void initEvaluator() throws DbException {
     ExpressionOperator op = getExpression().getRootExpressionOperator();
-
     String pyFunctionName = ((PyUDFExpression) op).getName();
 
     try {
       if (pyFuncRegistrar != null) {
         FunctionStatus fs = pyFuncRegistrar.getFunctionStatus(pyFunctionName);
         if (fs != null && fs.getName() == null) {
-          String pyCodeString = fs.getBinary();
-
-          if (pyCodeString == null) {
-            throw new DbException("No Python UDf with given name registered.");
-          } else {
-            if (pyWorker != null) {
-              isFlatmap = fs.getIsMultivalued(); //if the function is mutlivalued.
-              pyWorker.sendCodePickle(fs.getBinary(), numColumns, outputType, isFlatmap);
-            }
+          throw new DbException("No Python UDf with given name registered.");
+        } else {
+          if (pyWorker != null) {
```

Comment:
Before calling `initEvaluator`, you already gave `pyWorker` a value. It doesn't seem like we need this check.